mne/io/fiff/tests/test_raw.py

```diff
@@ -821,6 +821,59 @@ def test_resample():
     assert_equal(raw1.last_samp, raw3.last_samp)
     assert_equal(raw1.info['sfreq'], raw3.info['sfreq'])
 
+    # test resampling of stim channel
+
+    # basic decimation
+    stim = [1, 1, 1, 1, 0, 0, 0, 0, 1, 1, 1, 1, 0, 0, 0, 0]
+    raw = RawArray([stim], create_info(1, len(stim), ['stim']))
+    assert_allclose(raw.resample(8)._data,
```

Comment:
8 is not a float. 8 is an integer.

`raw.resample` should be able to deal with integers.